avalanche/training/utils.py

```diff
@@ -45,7 +45,10 @@ def at_task_boundary(training_experience) -> bool:
 
     if isinstance(training_experience, OnlineCLExperience):
         if training_experience.access_task_boundaries:
-            if training_experience.is_first_subexp:
+            if (
+                training_experience.is_first_subexp
+                or training_experience.is_last_subexp
+            ):

```

Comment:
can you explain this again? I'm a bit confused. With the current condition, we are going to update model and optimizer twice, at the final experience and at the new one.

Reply:
Ah, that's right. I had not thought about this. The idea is that this util function should be able to return both sides of the boundary. Maybe I should add an option to say which side to detect ?

For instance, I had the problem recently that to implement LwF for a compatibility with online, I also need to use this function but in the after_training_exp, so I need to know the right side of the boundary, whereas when it is called in before_training_exp I need to know the left side of the boundary.

Reply:
Ok, but make it private (`_at_task_boundary(before=True)`). It's ok as an internal utility but I think it would be a bit confusing for external users